src/main/scala/passes/LoopCheck.scala

```diff
@@ -0,0 +1,174 @@
+package fuselang.passes
+
+import scala.{PartialFunction => PF}
+
+//import fuselang.Utils.RichOption
+
+import fuselang.common._
+import ScopeMap._
+import ScopeMapExt._
+import Syntax._
+import Errors._
+import CompilerError._
+import Checker._
+import EnvHelpers._
+
+object LoopChecker {
+
+  def check(p: Prog) = LCheck.check(p)
+  object States extends Enumeration{
+    type States = Value
+    val USE, DEF, DK = Value
+  }
+  import States._
+
+  // Bounds checker environment doesn't need to track any information. Empty
+  // environment that just runs the commands.
+  private case class LEnv(
+    StateMap: ScopedMapExt[Id, States] = ScopedMapExt(),
```

Comment:
Field names and variable names in scale should [start with lowercase](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods).